stakedb/ticketpool.go

```diff
@@ -225,18 +175,20 @@ func NewTicketPool(dataDir, dbSubDir string) (tp *TicketPool, err error) {
 	// Open ticket pool diffs database
 	badgerDbPath := filepath.Join(dataDir, dbSubDir)
 	opts := badger.DefaultOptions(badgerDbPath)
+	opts.MetricsEnabled = false
 	opts.Logger = &badgerLogger{log}
 	db, err := badger.Open(opts)
-	if err == badger.ErrTruncateNeeded {
-		log.Warnf("NewTicketPool badger db: %v", err)
-		// Try again with value log truncation enabled.
-		opts.Truncate = true
-		log.Warnf("Attempting to reopening ticket pool db with the Truncate option set...")
-		db, err = badger.Open(opts)
-	}
 	if err != nil {
-		return nil, fmt.Errorf("failed badger.Open: %v", err)
+		if !strings.Contains(err.Error(), "manifest has unsupported version") {
+			return nil, fmt.Errorf("failed to open badger DB: %w", err)
+		}
```

Comment:
afaict, `ErrTruncateNeeded` still applies to v3 badger

Reply:
It only affects a `ReadOnly`  DB, badger will auto truncate.

Reply:
Where does it say that?  Did this change with v3?  It did not auto-truncate before.  Why else is this a flag?

Reply:
Oh, I see it was removed: https://github.com/dgraph-io/badger/pull/1555

Try to call out important API and functional changes that we are adapting to in the future.

Reply:
> Did this change with v3?

Yes, the `opts.Truncate` no longer exists on v3.

Reply:
> Did this change with v3?

Yes, the `opts.Truncate` no longer exists on v3.

Reply:
> Oh, I see it was removed: [dgraph-io/badger#1555](https://github.com/dgraph-io/badger/pull/1555)
> 
> Try to call out important API and functional changes that we are adapting to in the future.

Yeah, was about to drop a link when you asked but was disconnected from the internet.